src/Nest/Mapping/Types/FieldType.cs

```diff
@@ -69,5 +61,35 @@ public enum FieldType
 		Murmur3Hash,
 		[EnumMember(Value = "token_count")]
 		TokenCount,
+		[EnumMember(Value = "percolator")]
+		Percolator,
+
+		[EnumMember(Value = "integer")]
+		Integer,
+		[EnumMember(Value = "long")]
+		Long,
+		[EnumMember(Value = "short")]
+		Short,
+		[EnumMember(Value = "byte")]
+		Byte,
+		[EnumMember(Value = "float")]
+		Float,
+		[EnumMember(Value = "half_float")]
+		HalfFloat,
+		[EnumMember(Value = "scaled_float")]
+		ScaledFloat,
+		[EnumMember(Value = "double")]
+		Double,
+
+		[EnumMember(Value = "integer_range")]
+		IntegerRange,
+		[EnumMember(Value = "float_range")]
+		FloatRange,
+		[EnumMember(Value = "long_range")]
+		LongRange,
+		[EnumMember(Value = "double_range")]
+		DoubleRange,
+		[EnumMember(Value = "date_range")]
+		DateRange,
```

Comment:
This also needs to handle custom field types e.g. [Rosette's `name` data type](https://www.rosette.com/elastic/)